src/analysisd/eventinfo.c

```diff
@@ -25,6 +25,62 @@ EventList *last_events_list;
 int num_rule_matching_threads;
 time_t current_time = 0;
 
+size_t field_offset[] = { 
+    offsetof(Eventinfo, dstip),
+    offsetof(Eventinfo, srcport),
+    offsetof(Eventinfo, dstport),
+    offsetof(Eventinfo, srcuser),
+    offsetof(Eventinfo, dstuser),
+    offsetof(Eventinfo, protocol),
+    offsetof(Eventinfo, action),
+    offsetof(Eventinfo, url),
+    offsetof(Eventinfo, data),
+    offsetof(Eventinfo, extra_data),
+    offsetof(Eventinfo, status),
+    offsetof(Eventinfo, systemname),
+    offsetof(Eventinfo, srcgeoip),
+    offsetof(Eventinfo, dstgeoip),
+    offsetof(Eventinfo, location),
+    offsetof(Eventinfo, srcip),
+    offsetof(Eventinfo, id)
+};
+
+bool same_loop(RuleInfo *rule, Eventinfo *lf, Eventinfo *my_lf) {
+    int i;
+    u_int32_t same = rule->same_field;
+
+    for (i = 0; i < N_FIELDS-2 || same != 0; i++) {
```

Comment:
We can optimize the condition removing `i < N_FIELDS - 2`.
Each cycle of the loop will shift `same`, that's like dividing it by 2, so `same` will be set to 0 when it contains no more bits.